app/models/task.py

```diff
@@ -2,4 +2,41 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String) #when refactoring change to varchar
+    description = db.Column(db.String) # when refacoring change to varvhar
+    completed_at = db.Column(db.DateTime, default=None, nullable=True) 
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'))
+    goal = db.relationship("Goal", back_populates="tasks")
```

Comment:
Nice approach creating this relationship to your Goal model. Why not use `lazy` here?